### PR TITLE
small hack to change favicon based on tree status.

### PR DESCRIPTION
**Pull request by codebird**
Hack to change favicon based on tree status.

Closes flutter/flutter#46175

### CODE REVIEW

**Comment by codebird** [comment]
@jmagman @CaseyHillers would this pass?

File: dashboard/web/index.html [hunk 1]
```diff
@@ -24,17 +24,34 @@
...
+    js.src='https://apis.google.com/js/platform.js';
+    fjs.parentNode.insertBefore(js,fjs);js.onload=function(){g.load('analytics')};
+  }(window,document,'script'));
+
+  function checkTreeStatus(){
+    var link = document.querySelector("link[rel~='icon']");
+    if(window.location.href.indexOf('/build') > 0) {
+      glass = document.body.getElementsByTagName('flt-glass-pane')[0]
+
+      if (glass.getInnerHTML().indexOf('Tree is Open') > 0) {
```
[the remark this answers is not in the document]
Comment (codebird) [reply dashboard/web/index.html:44]: I am just searching for tree is open in all the html of flt-glass... Couldn't get into specific container. It is literally a hack

**Comment by CaseyHillers** [comment]
@yjbanov is there a way we could do this in Flutter?

**Comment by codebird** [comment]
After researching a bit around. I found that flutter doesn't do favicon, everywhere I looked it said place the icon in the web directory and update index.html that's why I went for js

File: dashboard/web/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply dashboard/web/index.html:44]: Ok. Can we update the favicons to not be the GitHub ones? When I looked at this, I was going to use the existing favicon, but tint it red for tree closures

[the remark this answers is not in the document]
Comment (codebird) [reply dashboard/web/index.html:44]: Sure, can I have the image. You don't want to see my graphical powers.

[the remark this answers is not in the document]
Comment (CaseyHillers) [reply dashboard/web/index.html:44]: failure

[the remark this answers is not in the document]
Comment (codebird) [reply dashboard/web/index.html:44]: Thanks, will update tomorrow.

File: dashboard/web/index.html [hunk 2]
```diff
@@ -37,18 +37,15 @@
...
+      if (glass.getInnerHTML().indexOf('Tree is Closed') > 0) {
+        icon = 'favicon-failure.png';
       }
-    } else {
-      link.href = 'favicon.png';
     }
+    link.href = icon;
     setTimeout(checkTreeStatus, 2000);
   }
   setTimeout(checkTreeStatus, 5000);
```
Comment (CaseyHillers) [inline dashboard/web/index.html:51]: Any reason for `setTimeout` over `setInterval`? It seems like that would let us remove the recursion

Comment (codebird) [reply dashboard/web/index.html]: just bad habit.

**Review by CaseyHillers — APPROVED** [review]
LGTM! Thanks for the feature, this will be nice to have!

Let me find a second flutter hacker whose willing to review javascript :-)

File: dashboard/web/index.html [hunk 3]
```diff
@@ -24,17 +24,30 @@
...
+    canvasKitBaseUrl: "/canvaskit/"
+  };
+  (function(w,d,s,g,js,fjs){
+    g=w.gapi||(w.gapi={});g.analytics={q:[],ready:function(cb){this.q.push(cb)}};
+    js=d.createElement(s);fjs=d.getElementsByTagName(s)[0];
+    js.src='https://apis.google.com/js/platform.js';
+    fjs.parentNode.insertBefore(js,fjs);js.onload=function(){g.load('analytics')};
+  }(window,document,'script'));
+
+  function checkTreeStatus(){
```
Comment (christopherfujino) [inline dashboard/web/index.html:39]: Can we use `dart:html` from Flutter for Web? It feels wrong doing this from JS. @CaseyHillers do you know?

File: dashboard/web/index.html [hunk 4]
```diff
@@ -24,17 +24,30 @@
...
+    let link = document.querySelector("link[rel~='icon']");
+    let icon = 'favicon.png';
+    if(window.location.href.indexOf('/build') > 0) {
```
Comment (christopherfujino) [inline dashboard/web/index.html:42]:
```suggestion
    if (window.location.href.indexOf('/build') > 0) {
```

Comment (codebird) [reply dashboard/web/index.html]: good eye!

**Comment by CaseyHillers** [comment]
As a future yak shave, ideally there'd be a Flutter plugin that supports setting the favicon. I wasn't able to find one in the pub ecosystem other than some that are meant to get one.

**Comment by codebird** [comment]
Yes I saw many to fetch icons. Now for most use cases you don't need to control the favicon, like it's static and that's it, unless for cases like this one. I don't know how frequent this is...

**Comment by codebird** [comment]
After giving it a thought I think it is frequent enough. I'll try to write a plugin just for the fun of it.

**Pull request merged**